s2spy/preprocess.py

```diff
@@ -0,0 +1,217 @@
+from typing import Tuple
+from typing import Union
+import numpy as np
+import scipy.stats
+import xarray as xr
+
+
+def _linregress(x: np.ndarray, y: np.ndarray) -> Tuple[float, float]:
+    """Calculate the slope and intercept between two arrays using scipy's linregress.
+
+    Used to make linregress more ufunc-friendly.
+
+    Args:
+        x: First array.
+        y: Second array.
+
+    Returns:
+        slope, intercept
+    """
+    slope, intercept, _, _, _ = scipy.stats.linregress(x, y)
+    return slope, intercept
+
+
+def _trend_linear(data: Union[xr.DataArray, xr.Dataset]) -> dict:
+    """Calculate the linear trend over time.
+
+    Args:
+        data: The input data of which you want to know the trend.
+
+    Returns:
+        Dictionary containing the linear trend information (slope and intercept)
+    """
+    slope, intercept = xr.apply_ufunc(
+        _linregress,
+        data["time"].astype(float),
```

Comment:
this astype(float) is a super nice solution! It easily allows to apply the trend to another (out-of-sample) dataset. This is something we were really struggling with @semvijverberg

Reply:
In principle I did a similar thing, but then the floating number was always anchored at 1970-01-01 (this is following the UNIX time definition).

Reply:
> but then the floating number was always anchored at 1970-01-01 

That is exactly what this does, assuming the format of data["time"] is a np.datetime64, which is how timestamps are stored in xarray data.

We should probably add a check that the time coordinate does indeed have timestamps.